src/utils/tokenSwap.js

```diff
@@ -0,0 +1,230 @@
+import { get } from 'lodash';
+
+/**
+ * Definition of a swap step.
+ * Multiple swap steps will configure the token swap path.
+ *
+ * @typedef {Object} TokenSwapPathStep
+ * @property {string} tokenIn Token being traded in
+ * @property {string} tokenOut Token being traded out
+ * @property {number} fee Fee charged for this swap
+ */
+
+/**
+ * Parsed token swap quote.
+ *
+ * @typedef {Object} TokenSwapQuote
+ * @property {TokenSwapPathStep[]} path Path the token swap will take.
+ * @property {string} pathStr Value returned by the contract before parsing.
+ * @property {number[]} amounts Value of the swap at each step.
+ * @property {number} amount Amount to be withdrawn or deposited, depending on the direction.
+ * @property {number} price_impact percentage value of how the swap will affect the price (with precision).
+ */
+
+/**
+ * Parse a quote path step from the string value.
+ * @param {string} pathStep
+ * @returns {TokenSwapPathStep}
+ */
+function parseQuotePathStep(pathStep) {
+  const step = pathStep.split('/');
+  step[2] = Number(step[2]);
+  return {
+    tokenIn: step[0],
+    tokenOut: step[1],
+    fee: Number(step[2]),
+  };
+}
+
+/**
+ * Fetch the token swap information for the proposed swap.
+ * The swap being proposed will deposit `amount` of `tokenIn` to the `contractId`.
+ * This method will find the best path and value of `tokenOut` we can withdraw given the amount deposited.
+ *
+ * @param {string} contractId The pool manager to find the swap
+ * @param {bigint} amount Amount to withdraw from the contract
+ * @param {string} tokenIn Token UID of the deposited token
+ * @param {string} tokenOut Token UID of the withdrawn token
+ * @returns {Promise<TokenSwapQuote>}
+ */
+async function findBestTokenSwapPathIn(contractId, amount, tokenIn, tokenOut) {
+  const call = `find_best_swap_path(${amount},"${tokenIn}","${tokenOut}",3)`;
+
+  const response = await ncApi.getNanoContractState(contractId, [], [], [call]);
+
+  if (!(response.calls && response.calls[call])) {
+    throw new Error('Did not receive any return data');
+  }
+  const data = response.calls[call];
+  return {
+    path: data.path.split(',').map(step => parseQuotePathStep(step)),
+    pathStr: data.path,
+    amounts: data.amounts,
+    amount: data.amount_out,
+    price_impact: data.price_impact,
+  };
+}
+
+/**
+ * Fetch the token swap information for the proposed swap.
+ * The swap being proposed will withdraw `amount` of `tokenOut` from the `contractId`.
+ * This method will find the best path and value to deposit of `tokenIn` to make it happen.
+ *
+ * @param {string} contractId The pool manager to find the swap
+ * @param {bigint} amount Amount to withdraw from the contract
+ * @param {string} tokenIn Token UID of the deposited token
+ * @param {string} tokenOut Token UID of the withdrawn token
+ * @returns {Promise<TokenSwapQuote>}
+ */
+async function findBestTokenSwapPathOut(contractId, amount, tokenIn, tokenOut) {
```

Comment:
This method and previous one are exactly the same. Maybe we could have a single method for it that receives the call. What do you think?